addons/sys_intercom/XEH_postInit.sqf

```diff
@@ -3,14 +3,76 @@
 // Exit if ACE3 not loaded
 if (!isClass (configFile >> "CfgPatches" >> "ace_interact_menu")) exitWith {};
 
-["Tank", "init", FUNC(intercomConfig), nil, nil, true] call CBA_fnc_addClassEventHandler;
-["Car_F", "init", FUNC(intercomConfig), nil, nil, true] call CBA_fnc_addClassEventHandler;
-["Air", "init", FUNC(intercomConfig), nil, nil, true] call CBA_fnc_addClassEventHandler;
-["Boat_F", "init", FUNC(intercomConfig), nil, nil, true] call CBA_fnc_addClassEventHandler;
+["Tank", "init", FUNC(initVehicleIntercom), nil, nil, true] call CBA_fnc_addClassEventHandler;
+["Car_F", "init", FUNC(initVehicleIntercom), nil, nil, true] call CBA_fnc_addClassEventHandler;
+["Air", "init", FUNC(initVehicleIntercom), nil, nil, true] call CBA_fnc_addClassEventHandler;
+["Boat_F", "init", FUNC(initVehicleIntercom), nil, nil, true] call CBA_fnc_addClassEventHandler;
 
 if (!hasInterface) exitWith {};
 
-ADDPFH(DFUNC(vehicleCrewPFH), 1.1, []);
+["vehicle", {
+    params ["_player", "_newVehicle"];
+
+    if (_player != _newVehicle) then {
+        _player setVariable[QGVAR(intercomVehicle), _newVehicle];
+
+        // Configure the array for handling if a player is in a limited intercom position or not.
+         private _usingLimitedPosition = [];
+        {
+            _usingLimitedPosition pushBack false;
+        } forEach (_newVehicle getVariable [QGVAR(intercomNames), []]);
+        player setVariable [QGVAR(usingLimitedPosition), _usingLimitedPosition];
+
+        {
+            [_newVehicle, _player, _forEachIndex] call FUNC(seatSwitched);
+        } forEach (_newVehicle getVariable [QGVAR(intercomNames), []]);
+        GVAR(crewPFH) = [DFUNC(intercomPFH), 1.1, [_player, _newVehicle]] call CBA_fnc_addPerFrameHandler;
+    } else {
+        [GVAR(crewPFH)] call CBA_fnc_removePerFrameHandler;
+        private _intercomVehicle = _player getVariable[QGVAR(intercomVehicle), objNull];
+        private _unitsIntercom = _intercomVehicle getVariable[QGVAR(unitsIntercom) , []];
+        private _disconnected = false;
+        {
+            private _temp = +_x;
+            if (_player in _temp) then {
+                _temp = _temp - [_player];
```

Comment:
I believe this is faster (also won't error in case you wonder, `find` returns `-1` if not found, which then doesn't delete anything):
```sfq
_temp deleteAt (_temp find _player);
```